packages/connect-partykit/src/server.ts

```diff
@@ -19,48 +27,31 @@ export default class Server implements Party.Server {
   }
 
   async onRequest(request: Party.Request) {
-    const CORS = {
-      'Access-Control-Allow-Origin': '*',
-      'Access-Control-Allow-Methods': 'GET, POST, OPTIONS, PUT,  DELETE',
-    }
-
-    const json = <T>(data: T, status = 200) =>
-      Response.json(data, { status, headers: CORS })
-
-    const ok = () => json({ ok: true })
-
-    // Check if it's a preflight request (OPTIONS) and handle it
     if (request.method === 'OPTIONS') {
       return ok()
     }
 
-    const url = new URL(request.url);
+    const url = new URL(request.url)
     const carId = url.searchParams.get('car')
 
     if (carId) {
       if (request.method === 'PUT') {
-        const carArrayBuffer = new Uint8Array(await request.arrayBuffer())
+        const carArrayBuffer = await request.arrayBuffer()
         if (carArrayBuffer) {
-          //Maybe add catch later?
           await this.party.storage.put(`car-${carId}`, carArrayBuffer)
           return json(JSON.stringify({ ok: true }), 201)
         }
         return json(JSON.stringify({ ok: false }), 400)
-      }
-       else if (request.method === 'GET') {
-        const carArrayBuffer = (await this.party.storage.get(
-          `car-${carId}`
-        )) as Uint8Array;
+      } else if (request.method === 'GET') {
+        const carArrayBuffer = (await this.party.storage.get(`car-${carId}`)) as Uint8Array
         if (carArrayBuffer) {
-          return json(carArrayBuffer);
+          return new Response(carArrayBuffer, { status: 200, headers: CORS })
```

Comment:
if the patch doesn't work, try reverting to `main` and just changing this one line @valorant-dhruv